http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -505,12 +513,217 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
       _ <- publisherQueue.offer(message)
     } yield ()
 
+  override def listAppV2(userInfo: UserInfo, workspaceId: WorkspaceId, params: Map[String, String])(implicit
+    as: Ask[F, AppContext]
+  ): F[Vector[ListAppResponse]] = for {
+    paramMap <- F.fromEither(processListParameters(params))
+    allClusters <- KubernetesServiceDbQueries
+      .listAppsByWorkspaceId(Some(workspaceId), paramMap._1, paramMap._2)
+      .transaction
+
+    res = allClusters
+      .flatMap(c => ListAppResponse.fromCluster(c, Config.proxyConfig.proxyUrlBase, paramMap._3))
+      .toVector
+    // TODO: Figure out permissions
+    // See listApp for old implementation.
+  } yield res
+
+  override def getAppV2(userInfo: UserInfo, workspaceId: WorkspaceId, appName: AppName)(implicit
+    as: Ask[F, AppContext]
+  ): F[GetAppResponse] =
+    for {
+      ctx <- as.ask
+      appOpt <- getActiveFullAppByWorkspaceIdAndAppName(workspaceId, appName).transaction
+      app <- F.fromOption(
+        appOpt,
+        AppNotFoundByWorkspaceIdException(workspaceId, appName, ctx.traceId, "No active app found in DB")
+      )
+      hasPermission <- authProvider.hasPermission[AppSamResourceId, AppAction](app.app.samResourceId,
+                                                                               AppAction.GetAppStatus,
+                                                                               userInfo
+      )
+      _ <-
+        if (hasPermission) F.unit
+        else
+          log.info(ctx.loggingCtx)(
+            s"User ${userInfo} tried to access app ${appName.value} without proper permissions. Returning 404"
+          ) >> F
+            .raiseError[Unit](AppNotFoundByWorkspaceIdException(workspaceId, appName, ctx.traceId, "permission denied"))
+    } yield GetAppResponse.fromDbResult(app, Config.proxyConfig.proxyUrlBase)
+
+  override def createAppV2(userInfo: UserInfo, workspaceId: WorkspaceId, appName: AppName, req: CreateAppRequest)(
+    implicit as: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- as.ask
+      userToken = org.http4s.headers.Authorization(
+        org.http4s.Credentials.Token(AuthScheme.Bearer, userInfo.accessToken.token)
+      )
+      workspaceDescOpt <- wsmDao.getWorkspace(workspaceId, userToken)
+      workspaceDesc <- F.fromOption(workspaceDescOpt, WorkspaceNotFoundException(workspaceId, ctx.traceId))
+
+      hasPermission <- authProvider.hasPermission[WorkspaceResourceSamResourceId, WorkspaceAction](
+        WorkspaceResourceSamResourceId(workspaceId),
+        WorkspaceAction.CreateControlledUserResource,
```

Comment:
`WorkspaceAction.CreateControlledUserResource` seems like the right thing to check here (even though we're not actually making a controlled resournce ;))